recipes/tar/all/conanfile.py

```diff
@@ -1,98 +1,112 @@
             "--disable-nls",
             "--disable-rpath",
-            # "--without-gzip",  # FIXME: this will use system gzip
             "--without-posix-acls",
             "--without-selinux",
-            "--with-bzip2={}".format(bzip2_exe),
-            "--with-lzip={}".format(lzip_exe),
-            "--with-lzma={}".format(lzma_exe),
-            # "--without-lzop",  # FIXME: this will use sytem lzop
-            "--with-xz={}".format(xz_exe),
-            # "--without-zstd",  # FIXME: this will use system zstd (current zstd recipe does not build programs)
+            "--with-gzip=gzip",  # FIXME: this will use system gzip
+            "--with-bzip2=bzip2",
+            "--with-lzip=lzip",
+            "--with-lzma=lzma",
+            "--without-lzop", # FIXME: lzo package does not build an executable
+            "--with-xz=xz",
+            "--with-zstd=zstd",
         ]
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        if is_msvc(self):
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "gnu", "faccessat.c"),
+                "_GL_INCLUDING_UNISTD_H",
+                "_GL_INCLUDING_UNISTD_H_NOP",
+            )
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        if self.settings.compiler == "Visual Studio":
-            tools.replace_in_file(os.path.join(self._source_subfolder, "gnu", "faccessat.c"),
-                                  "_GL_INCLUDING_UNISTD_H", "_GL_INCLUDING_UNISTD_H_NOP")
-        autotools = self._configure_autotools()
+        self._patch_sources()
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        autotools = Autotools(self)
         autotools.install()
-
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "libexec"))
 
     def package_info(self):
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        self.cpp_info.frameworkdirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+        self.cpp_info.includedirs = []
 
         tar_bin = os.path.join(self.package_folder, "bin", "tar")
-        self.user_info.tar = tar_bin
+        self.conf_info.define("user.tar:tar", tar_bin)
```

Comment:
```suggestion
        self.conf_info.define("user.tar:path", tar_bin)
```
A bit more specific about the which property of tar